Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -114,44 +170,36 @@ protected override bool IsValidController(IMixedRealityController controller)
+                    {
+                        if (followHandUntilFacingCamera)
                         {
-                            var handNormal = Vector3.Cross(indexTipPose.Position - palmPose.Position,
-                                                           ringTipPose.Position - indexTipPose.Position).normalized;
-                            handNormal *= (jointedHand.ControllerHandedness == Handedness.Right) ? 1.0f : -1.0f;
-
-                            if (Vector3.Angle(palmPose.Up, handNormal) > flatHandThreshold)
+                            // If we are above the threshold angle, keep the menu mapped to the tracked object
+                            if (palmCameraAngle > followHandCameraFacingThresholdAngle)
+                            {
+                                RotationBehavior = SolverRotationBehavior.LookAtTrackedObject;
+                                OffsetBehavior = SolverOffsetBehavior.TrackedObjectRotation;
+                            }
+                            // If we are within the threshold angle, we snap to follow the camera
+                            else
                             {
-                                return false;
+                                RotationBehavior = SolverRotationBehavior.LookAtMainCamera;
+                                OffsetBehavior = SolverOffsetBehavior.LookAtCameraRotation;
                             }
                         }
-                    }
 
-                    // Check if the palm angle meets the prescribed threshold
-                    palmFacingThresholdMet = palmCameraAngle < facingCameraTrackingThreshold;
-
-                    // If using hybrid hand rotation, we proceed with additional checks
-                    if (followHandUntilFacingCamera && palmFacingThresholdMet)
-                    {
-                        // If we are above the threshold angle, keep the menu mapped to the tracked object
-                        if (palmCameraAngle > followHandCameraFacingThresholdAngle)
+                        if (useGazeActivation && (!gazeActivationAlreadyTriggered || targetWorldLocked))
                         {
-                            RotationBehavior = SolverRotationBehavior.LookAtTrackedObject;
-                            OffsetBehavior = SolverOffsetBehavior.TrackedObjectRotation;
-                        }
-                        // If we are within the threshold angle, we snap to follow the camera
-                        else
-                        {
-                            RotationBehavior = SolverRotationBehavior.LookAtMainCamera;
-                            OffsetBehavior = SolverOffsetBehavior.LookAtCameraRotation;
+                            return IsUserGazeMeetingThresholdRequirements(jointedHand);
                         }
                     }
+
+                    gazeActivationAlreadyTriggered = palmFacingThresholdMet ? gazeActivationAlreadyTriggered : false;
```

Comment:
I had to think about this for a second...to make the code more readable, what do you think about:

```
if (!palmFacignThresholdMet) {
  gazeActivationAlreadyTriggered = false;
}
```

It's minor but that tells me more what I think this is doing already (reset the variable if palm isn't facing you)

Reply:
Sure; I can make that change